src/Storages/StorageS3.cpp

```diff
@@ -893,6 +896,13 @@ Pipe StorageS3::read(
         block_for_format = storage_snapshot->metadata->getSampleBlock();
     }
 
+    auto new_format_settings = format_settings;
```

Comment:
But now it will work only for `Parquet` and `Arrow` formats. I think that it will be more misleading than now if support of subcolumns will work only for a subset of formats. Also we have other special subcolumns like `.null` of `Nullable` types and `.size0` in `Array` types. They won't work as well.

Reply:
I think the support of subcolumns can be done by requesting the full column and extracting subcolumn by calling of function `getColumnFromBlock` ([link](https://github.com/ClickHouse/ClickHouse/blob/0b4e643c2764543911b0b4c0c3745846af05742e/src/Interpreters/getColumnFromBlock.h#L9)).

As I understand, options of formats like `import_nested` don't allow to read only requested nested columns, but instead allow to read the whole column and extract requested nested columns. So, the performance of suggested solution will be the same.